src/pal/src/misc/miscpalapi.cpp

```diff
@@ -331,9 +306,9 @@ PAL_Random(
     }
 
     // always xor srand48 over the whole buffer to get some randomness
-    // in case /dev/random is not really random
+    // in case /dev/urandom is not really random
 
-    for(i = 0; i < dwLength; i++)
+    for (i = 0; i < dwLength; i++)
```

Comment:
Can someone help me understand why this loop with xor'ing with mrand48 is necessary?  My understanding is that /dev/urandom will produce as much cryptographically-strong random data as it can and then fall back to pseudo-random data.  How does then mixing it with other prng data from mrand48 help?

Reply:
I don't think it would help. In fact, looking at the man page, it is not thread safe. The main reason why it was here was to work as a fallback in case the /dev/urandom is not present or cannot be opened for some reason.
I guess we can just say that the /dev/urandom is required to be present and if opening it fails, just abort the process. Then we can remove this loop.